packages/undo-manager/README.md

```diff
@@ -0,0 +1,33 @@
+# Undo Manager
+
+A simple undo manager.
```

Comment:
Maybe it's worth illustrating how this can be used with the block editor provider etc.?

Reply:
> The next step for me is going to be to write a React hook around this undo/redo manager, something like that:
> const { value, onChange, hasUndo, hasRedo, undo, redo } = useStateWithHistory( initialValue );

Actually, I'm planning to offer a wrapper that can be used with `BlockEditorProvider`. This package is a bit more lower level. it's a bit more complex and featured compared to what's needed for the block editor provider. In most cases, developers don't need multi-record + multi-property undo manager but it can serve as a low level abstraction for both use-cases.

So in that sense, I'll add the docs when I have the higher-level abstraction (we won't use the higher-level hook ourselves though)

Reply:
In that case, wouldn't it be more prudent to make this API private instead of exposing it immediately? It's fine if it's a package with just a private API? Then you can use this in the wrapper and expose only that for now.

Reply:
Personally, I'm kind of confident that this is a good package. Also, the only API it exports is `createUndoManager`. It's not an API folks would use unless they want to create their own undo/redo stack for their own stuff.

Reply:
If we really want to keep this package private on Core, we can make it a bundled package like "icons" or "interface". WDYT